mne/epochs.py

```diff
@@ -2201,25 +2207,139 @@ def read_epochs(fname, proj=True, add_eeg_ref=True, verbose=None):
     epochs : instance of Epochs
         The epochs
     """
-    check_fname(fname, 'epochs', ('-epo.fif', '-epo.fif.gz'))
-
-    fnames = [fname]
-    epochs = list()
-    for fname in fnames:
-        logger.info('Reading %s ...' % fname)
-        fid, tree, _ = fiff_open(fname)
-        next_fname = _get_next_fname(fid, fname, tree)
-        epoch = _read_one_epoch_file(fid, tree, fname, proj, add_eeg_ref,
-                                     verbose)
-        epochs.append(epoch)
-
-        if next_fname is not None:
-            fnames.append(next_fname)
-
-    epochs = _concatenate_epochs(epochs, read_file=True)
-    epochs._bad_dropped = True
+    return EpochsFIF(fname, proj, add_eeg_ref, preload, verbose)
 
-    return epochs
+
+class _RawContainer(object):
+    def __init__(self, fid, data_tag, event_samps, epoch_shape, cals):
+        self.fid = fid
+        self.data_tag = data_tag
+        self.event_samps = event_samps
+        self.epoch_shape = epoch_shape
+        self.cals = cals
+        self.proj = False
+
+
+class EpochsFIF(_BaseEpochs):
```

Comment:
Need to provide the `_get_epoch_from_raw(self, idx, verbose=None)` method. FWIW it actually makes more sense to me to have a different class on reading `-epo.fif` than creation of an `Epochs(raw, ...)` object. They provide the same data and the same interfaces, but how they operate under the hood is a bit different, which is how our other IO-style classes are (including `EpochsArray` and `EpochsKIT`).